src/bin/compute-wer-confidence.cc

```diff
@@ -0,0 +1,247 @@
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "util/parse-options.h"
+#include "tree/context-dep.h"
+#include "util/edit-distance.h"
+#include "base/kaldi-math.h"
+
+namespace kaldi {
+
+void GetEditsSingleHyp( const std::string &hyp_rspecifier, 
+      const std::string &ref_rspecifier,
+      const std::string &mode,
+      std::vector<std::pair<int32, int32> > & edit_word_per_hyp) {
+    
+    // Both text and integers are loaded as vector of strings,
+    SequentialTokenVectorReader ref_reader(ref_rspecifier);
+    RandomAccessTokenVectorReader hyp_reader(hyp_rspecifier);
+    int32 num_words = 0, word_errs = 0, num_ins = 0, num_del = 0, num_sub = 0;
+    
+    // Main loop, store WER stats per hyp,
+    for (; !ref_reader.Done(); ref_reader.Next()) {
+      std::string key = ref_reader.Key();
+      const std::vector<std::string> &ref_sent = ref_reader.Value();
+      std::vector<std::string> hyp_sent;
+      if (!hyp_reader.HasKey(key)) {
+        if (mode == "strict")
+          KALDI_ERR << "No hypothesis for key " << key << " and strict "
+              "mode specifier.";
+        if (mode == "present")  // do not score this one.
+          continue;
+      } else {
+        hyp_sent = hyp_reader.Value(key);
+      }
+      num_words = ref_sent.size();
+      word_errs = LevenshteinEditDistance(ref_sent, hyp_sent, 
+                                            &num_ins, &num_del, &num_sub);
+      edit_word_per_hyp.push_back(std::pair<int32, int32>(word_errs, num_words));
+    }
+}
+
+void GetEditsDualHyp(const std::string &hyp_rspecifier, 
+      const std::string &hyp_rspecifier2, 
+      const std::string &ref_rspecifier,
+      std::vector<std::pair<int32, int32> > & edit_word_per_hyp,
+      std::vector<std::pair<int32, int32> > & edit_word_per_hyp2) {
+    
+    // Both text and integers are loaded as vector of strings,
+    SequentialTokenVectorReader ref_reader(ref_rspecifier);
+    RandomAccessTokenVectorReader hyp_reader(hyp_rspecifier);
+    RandomAccessTokenVectorReader hyp_reader2(hyp_rspecifier2);
+    int32 num_words = 0, word_errs = 0, 
+            num_ins = 0, num_del = 0, num_sub = 0;
+    
+    // Main loop, store WER stats per hyp,
+    for (; !ref_reader.Done(); ref_reader.Next()) {
+      std::string key = ref_reader.Key();
+      const std::vector<std::string> &ref_sent = ref_reader.Value();
+      std::vector<std::string> hyp_sent, hyp_sent2;
+      if (!hyp_reader.HasKey(key) && !hyp_reader2.HasKey(key)) {
```

Comment:
This makes me a little bit unsure. If you use compute-wer after post-processing the outputs, the utterances might actually disappear (or just be completely empty). I wonder if you'd be willing to improve the code in similar way the original compute-wer is -- it has two or three modes how it deals with missing utterance.